x-pack/plugins/apm/public/components/app/correlations/correlations_chart.tsx

```diff
@@ -149,7 +145,28 @@ export function CorrelationsChart({
     },
   ];
 
-  const xMax = Math.max(...overallHistogram.map((d) => d.key)) ?? 0;
+  const xMax = Math.max(...(overallHistogram ?? []).map((d) => d.key)) ?? 0;
+
+  // We want y axis ticks for 1, 10, 100, 1000 ...
+  // First get the actual max of y values
+  const yMax =
+    Math.max(...(overallHistogram ?? []).map((d) => d.doc_count)) ?? 0;
+  // The axis treats the ticks number as a best effort/recommendation.
+  // By counting the digits of yMax we make sure it's one less than what we want.
+  // The axis will then round up the ticks and we won't overshoot with many tick values.
+  const yTicks = ('' + Math.round(yMax)).length;
+  // Up the maximum y domain value to the next nice log label,
+  // e.g. if the actual max is 234, we'll up it to 1000.
+  const yDomainMax = parseInt(
+    `1${new Array(('' + Math.round(yMax)).length).fill(0).join('')}`,
+    10
+  );
```

Comment:
This all seem rather manual. I'd rather avoid us having to think about this, and let d3 handle it if possible.

```js
d3.scaleLog().domain([1, 10000]).ticks(4) // [1, 10, 100, 1000, 10000]
```

I imagine we can then use it like:
```
d3.scaleLog().domain([1, yMax]).ticks(Math.log10(yMax))
```

There is a d3 sandbox here: https://observablehq.com/@d3/scale-ticks

Reply:
... Actually, perhaps I misunderstood, and we don't need to calculate the actual tick values.
Still, I'm not sure if the right tick values will be generated. Eg if we have a series that has a y max of 75 we want three ticks: 1, 10, 100.

Not sure if the current calculation will give us that but I think the following is a little more robust:

```ts
const yTicks = Math.ceil(Math.log10(yMax))
const yAxisDomain = {
  min: 1,
  max: Math.pow(10, yTicks),
};
```